app/src/main/java/com/example/a310_rondayview/ui/InterestedEventsAdapter.java

```diff
@@ -0,0 +1,70 @@
+package com.example.a310_rondayview.ui;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.example.a310_rondayview.Event;
+import com.example.a310_rondayview.R;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class InterestedEventsAdapter extends RecyclerView.Adapter<InterestedEventsAdapter.InterestedEventsViewHolder> {
+
+    Context context;
+    ArrayList<Event> eventsArrayList;
```

Comment:
Does this need to be an array list? Can we change it to List<Event> so we don't have to cast it later on?

Reply:
Sounds like a better idea, good catch will change that now :)